### ISSUE TITLE
Events Not Firing

### ISSUE THREAD

**Issue by sdoward** [roboguice/roboguice#260]
I have just upgraded to Roboguice 3.0. Great work, it seems super fast and fixes the performance problems we were having with ART. Unfortunately none of the events are working. Both custom events that I am firing and standard events such as onStopEvent and onStartEvent. 

I looked into the upgrade docs and it doesn't mention that I should change anything. Is this a known issue or do I need to change something?

**Comment by timroes** [comment]
I can confirm this bug. After half a day of disassembling and diffing files (with a correct build from JDK 1.7), it seems that some JDK version have a problem to correctly generate the `AnnotationDatabaseImpl` file. In the broken version the `fillAnnotationClassesAndMethods()` methods is completely empty (and some other stuff is missing too) which leads to events not working at all.

I updated my JDK form 1.8.0_05 to 1.8.0_45 and it started working. So for the case anyone is still stumble across this (as I did), update your JDK to the most recent versions and things seem to be fine.
